File: sdks/python/bolt_prediction_sdk/models/strategy_rollback_request.py
"""
StrategyRollbackRequest 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class StrategyRollbackRequest(SDKBaseModel):
    """策略回滚请求"""

    target_version: int = Field(description="回滚目标版本号")
    scope: Optional[str] = Field(description="作用域", default='global')
    node_type: Optional[Any] = Field(description="节点类型", default=None)
    node_id: Optional[Any] = Field(description="节点ID", default=None)
    operator_id: Optional[Any] = Field(description="操作人ID", default=None)
    operator_name: Optional[Any] = Field(description="操作人姓名", default=None)

File: sdks/java/src/main/java/com/boltprediction/sdk/model/StrategyRollbackRequest.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 策略回滚请求 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class StrategyRollbackRequest {

    @JsonProperty("target_version")
    private Integer targetVersion;

    @JsonProperty("scope")
    private String scope;

    @JsonProperty("node_type")
    private Object nodeType;

    @JsonProperty("node_id")
    private Object nodeId;

    @JsonProperty("operator_id")
    private Object operatorId;

    @JsonProperty("operator_name")
    private Object operatorName;

    public Integer getTargetVersion() {
        return targetVersion;
    }

    public void setTargetVersion(Integer targetVersion) {
        this.targetVersion = targetVersion;
    }

    public String getScope() {
        return scope;
    }

    public void setScope(String scope) {
        this.scope = scope;
    }

    public Object getNodeType() {
        return nodeType;
    }

    public void setNodeType(Object nodeType) {
        this.nodeType = nodeType;
    }

    public Object getNodeId() {
        return nodeId;
    }

    public void setNodeId(Object nodeId) {
        this.nodeId = nodeId;
    }

    public Object getOperatorId() {
        return operatorId;
    }

    public void setOperatorId(Object operatorId) {
        this.operatorId = operatorId;
    }

    public Object getOperatorName() {
        return operatorName;
    }

    public void setOperatorName(Object operatorName) {
        this.operatorName = operatorName;
    }

}